diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -74,7 +241,202 @@ class LineLength : Rule("line-length") {
+        text += node.findChildByType(RPAR)!!.text
+        return text
+    }
+
+    /**
+     * This method stored all the nodes that have BINARY_EXPRESSION or PREFIX_EXPRESSION element type.
+     * This method uses recursion to store binary node in the order in which they are located
+     * Also binList contains nodes with PREFIX_EXPRESSION element type ( !isFoo(), !isValid)
+     *@param node node in which to search
+     *@param binList mutable list of ASTNode to store nodes
+     */
+    private fun searchBinaryExpression(node: ASTNode, binList: MutableList<ASTNode>) {
+        if (node.hasChildOfType(BINARY_EXPRESSION) || node.hasChildOfType(PARENTHESIZED) ||
+                node.hasChildOfType(POSTFIX_EXPRESSION)) {
+            node.getChildren(null)
+                    .filter {
+                        it.elementType == BINARY_EXPRESSION || it.elementType == PARENTHESIZED ||
+                                it.elementType == POSTFIX_EXPRESSION
+                    }
+                    .forEach {
+                        searchBinaryExpression(it, binList)
+                    }
+        } else {
+            binList.add(node)
+            binList.add(node.treeParent.findChildByType(PREFIX_EXPRESSION) ?: return)
+        }
+    }
+
+    private fun dfsForProperty(node: ASTNode, binList: MutableList<ASTNode>) {
+        node.getChildren(null).forEach {
+            if (it.elementType in PROPERTY_LIST)
+                if (it.elementType == REFERENCE_EXPRESSION && it.treeParent.elementType == CALL_EXPRESSION)
+                    binList.add(it.treeParent)
+                else
+                    binList.add(it)
+            dfsForProperty(it, binList)
+        }
+    }
+
+    private fun fixLongString(fixableType: LongLineFixableEnum) {
+        val wrongNode = fixableType.node
+        val leftOffset = fixableType.leftOffset
+        val lineLength = fixableType.configuration.lineLength
+        val binList = fixableType.binList
+        val allText = binList.joinToString("") { it.text }
+        var binaryText = ""
+        binList.forEachIndexed { index, astNode ->
+            binaryText += astNode.text
+            if (STRING_PART_OFFSET + leftOffset + binaryText.length > lineLength) {
+                if (astNode.elementType == LITERAL_STRING_TEMPLATE_ENTRY) {
+                    val lastCharIndex = lineLength.toInt() - leftOffset - STRING_PART_OFFSET
+                    val indexLastSpace = (allText.substring(0, lastCharIndex).lastIndexOf(' '))
+                    if (indexLastSpace == -1)
+                        return
+                    splitTextAndCreateNode(wrongNode, allText, indexLastSpace)
+                    return
+                } else if (index == 0) {
+                    return
+                }
+                splitTextAndCreateNode(wrongNode, allText, binaryText.length)
+                return
+            }
+        }
+    }
+
+    private fun createNodes(fixableType: LongLineFixableEnum) {
```

Comment:
Please make name of this method more self-descriptive. It's strange to see construction like `LongLineFixableEnum.PROPERTY -> createNodes(fixableType)`.